Ginger/Ginger/ALM/ALMIntegration.cs

```diff
@@ -489,48 +489,86 @@ public ObservableList<ExternalItemFieldBase> GetALMItemFieldsREST(bool online, A
-                       
-                        if (latestField.Mandatory == true)
-                            existingField.ToUpdate = true;
-
-                        existingField.PossibleValues = latestField.PossibleValues;
-                        if (string.IsNullOrEmpty(existingField.SelectedValue) == false)
-                        {                        
-                            if (!latestField.PossibleValues.Contains(existingField.SelectedValue))
-                                existingField.SelectedValue = latestField.SelectedValue;
+                        currentField.Name = latestField.Name;
+                        currentField.ItemType = latestField.ItemType;
+                        currentField.Mandatory = latestField.Mandatory;
+                        currentField.ExternalID = latestField.ExternalID;
+                        currentField.PossibleValues = latestField.PossibleValues;
+                        currentField.ToUpdate = false;
+                        if (string.IsNullOrEmpty(currentField.SelectedValue) == false)
+                        {
+                            if ((latestField.PossibleValues.Count == 0 && currentField.SelectedValue != latestField.SelectedValue) || (latestField.PossibleValues.Count > 0 && latestField.PossibleValues.Contains(currentField.SelectedValue) && currentField.SelectedValue != latestField.PossibleValues[0]))
+                            {
+                                currentField.ToUpdate = true;
+                            }
+                            else
+                            {
+                                currentField.SelectedValue = latestField.SelectedValue;
+                                currentField.ToUpdate = false;
+                            }
                         }
                         else
                         {
-                            existingField.SelectedValue = latestField.SelectedValue;
+                            currentField.SelectedValue = latestField.SelectedValue;
+                            currentField.ToUpdate = false;
                         }
+                        mergedFields.Add(currentField);
                     }
                     else
-                        exitingFields.Add(latestField);
+                    {
+                        mergedFields.Add(latestField);
+                    }
                 }
 
-                //Sorting the list
-                ObservableList<ExternalItemFieldBase> sortedFields = new ObservableList<ExternalItemFieldBase>(from i in exitingFields orderby i.ItemType select i);
                 exitingFields.ClearAll();
-                exitingFields.Append(sortedFields);
+                exitingFields.Append(mergedFields);
+            }
+
+        }
+        internal ObservableList<ExternalItemFieldBase> getUpdatedFields(ObservableList<ExternalItemFieldBase> mItemsFields, bool online, BackgroundWorker bw = null)
+        {
```

Comment:
Function name getUpdatedFields should start with Capital letter like : "GetUpdatedFields "